server/player.lua

```diff
@@ -1189,14 +1189,17 @@ exports('SetCharInfo', SetCharInfo)
 ---@param moneyType MoneyType
 ---@param amount number
 ---@param actionType 'add' | 'remove' | 'set'
----@param direction boolean
+---@param difference number
 ---@param reason? string
-local function emitMoneyEvents(source, playerMoney, moneyType, amount, actionType, direction, reason)
-    TriggerClientEvent('hud:client:OnMoneyChange', source, moneyType, amount, direction)
+local function emitMoneyEvents(source, playerMoney, moneyType, amount, actionType, difference, reason)
+    local isSet = actionType == 'set'
+    local isRemove = actionType == 'remove'
+
+    TriggerClientEvent('hud:client:OnMoneyChange', source, moneyType, isSet and difference or amount, isSet and difference < 0 or isRemove and true or false, reason)
```

Comment:
```suggestion
    TriggerClientEvent('hud:client:OnMoneyChange', source, moneyType, isSet and difference or amount, isSet and difference < 0 or isRemove, reason)
```

Reply:
difference will never be less than 0 since we're passing in the absolute value. No behavior change is desired compared to qb-core's spec for this event https://github.com/qbcore-framework/qb-core/blob/main/server/player.lua#L377

Reply:
ok i passed raw the difference then we switch it to absolute value

Reply:
also, should i make the difference param the latest, and just pass it as nil for `RemoveMoney` and `AddMoney`

Reply:
Not sure what you mean by that

Reply:
i did change `difference` param order, and make it the last

Reply:
Oh, I see. Yeah that's fine.